### DISCUSSION TITLE
Beam support

### DISCUSSION THREAD

**Discussion by jbadeau** [AbsaOSS/spline-spark-agent#279] [Q&A]
## Question
As Apache Beam can run pipelines on Spark. Will Spline agents work if the beam pipelines are run on Spark?

**Comment by wajda** [comment]
I don't know how _Beam_ works on Spark under the hood, but here's some thoughts:

The Spline agent captures _DataFrame_ level transformations by listening to Spark query execution events emitted from write commands.
So in theory, as long as _Beam_ utilizes _DataFrame_'s and calls one of the [supported write commands](https://github.com/AbsaOSS/spline-spark-agent#implemented) then it should be supported out of the box.

If Beam implements a custom write command, but still utilizes `DataFrame` API then a [plugin](https://github.com/AbsaOSS/spline-spark-agent#plugins) will be required to capture the lineage.

The worst case would be if _Beam_ works on a RDD level, that is not supported at the moment (see #33).
